docs/components/index.md

```diff
@@ -0,0 +1,360 @@
+```
+
+When the check box is selected and cleared, the property's value is updated to `true` and `false`, respectively.
+
+The check box is updated in the UI only when the component is rendered, not in response to changing the property's value. Since components render themselves after event handler code executes, property updates are usually reflected in the UI immediately.
+
+Using `bind` with a `CurrentValue` (`<input bind="@CurrentValue" />`) is essentially equilivant to the following:
+
+```cshtml
+<input value="@CurrentValue" onchange="@((UIValueEventArgs __e) => CurrentValue = __e.Value) />
+```
+
+When the component is rendered, the `value` of the input element comes from the `CurrentValue` property. When the user types in the text box, the `onchange` is fired and the `CurrentValue` property is set to the changed value. In reality, the code generation is a little more complex because `bind` deals with a few cases where type conversions are performed. In principle, `bind` associates the current value of an expression with a `value` attribute and handles changes using the registered handler.
+
+**Format strings**
+
+Two-way databinding works with [DateTime](https://docs.microsoft.com/dotnet/api/system.datetime) format strings (but not other format expressions at this time, such as currency or number formats):
+
+```cshtml
+<input bind="@StartDate" format-value="yyyy-MM-dd" />
+
+@functions {
+    public DateTime StartDate { get; set; } = 
+        new DateTime(2020, 1, 1);
+}
+```
+
+**Conditional attributes**
+
+Blazor conditionally renders attributes based on the bound .NET value. If the value bound is `false` or `null`, Blazor won't render the attribute. If the value is `true`, the attribute is rendered minimized.
+
+In the following example, `IsCompleted` determines if `checked` is rendered in the control's markup.
+
+```cshtml
+<input type="checkbox" checked="@IsCompleted" />
+
+@functions {
+    public bool IsCompleted { get; set; }
+}
+```
+
+If `IsCompleted` is `true`, the check box is rendered as:
+
+```html
+<input type="checkbox" checked />
+```
+
+If `IsCompleted` is `false`, the check box is rendered as:
+
+```html
+<input type="checkbox" />
+```
+
+**Component attributes**
+
+Binding also recognizes component attributes, where `bind-{property}` can bind a property value across components.
+
+The following parent component uses `ChildComponent` and binds the value `1979` from `ParentYear` to the child component's `Year` property:
+
+Parent component:
+
+```cshtml
+<ChildComponent bind-Year="@ParentYear" />
+
+@functions {
+    public int ParentYear { get; set; } = 1979;
+}
+```
+
+Child component:
+
+```cshtml
+<div> ... </div>
+
+@functions {
+    public int Year { get; set; }
+    public Action<int> ValueChanged { get; set; }
+}
+```
+
+## Event handling
+
+Blazor offers two event handling features, `onclick` and `onchange`. The following code calls the `UpdateHeading` method when the button is selected in the UI:
```

Comment:
We now handle arbitrary events, not just `onclick` and `onchange`. In general the rule is: if you have an attribute named `onX`, and supply a delegate-typed value, then we treat it as an event handler for the event named `X`. For example, `onclick`, `onsubmit`, etc.

An easier way to explain this is just to say that to handle an event, developers assign a value to an attribute named after the event. The attribute's name must always start with `on`.

Reply:
Also the word "selected" should be "clicked".

Reply:
We must say "select"/"selected" given that Blazor apps run on mobile devices and tablets (no mouse) ... it's MS style manual guidance.